src/Hadrian/Package/Type.hs

```diff
@@ -0,0 +1,45 @@
+module Hadrian.Package.Type where
```

Comment:
From a quick look at `Hadrian.Package`, it doesn't seem that this split is necessary. Or is it?

Reply:
I think it will be necessary shortly (as in, in a PR or two), and given that we're not really in a comparable situation as with the `Stage`/`Stage.Type` case (the `.Package` module has several functions and the package-related types are likely going to be needed by "base" modules), I would personally suggest that we leave that one as-is, but it's your call of course :)